src/lfo.rs

```diff
@@ -0,0 +1,109 @@
+use crate::ring_buffer::RingBuffer;
+use std::f32::consts::PI;
+
+pub struct LFO {
+    wave_table: RingBuffer<f32>,
+    sample_rate_hz: f32,
+    phase_index: f32,
+    freq_hz: f32,
+    amplitude: f32,
+}
+
+impl LFO {
+    pub fn new(sample_rate_hz: f32, size: usize) -> Self {
+        // size determine the resolution of the wave table
+        let mut wave_table = RingBuffer::<f32>::new(size);
+        for i in 0..size {
+            let phase = (i as f32 / size as f32) * 2.0 * PI;
+            wave_table.push(phase.sin());
+        }
+        wave_table.push(0.0);
+        Self {
+            wave_table,
+            sample_rate_hz,
+            freq_hz: 0.0,
+            amplitude: 1.0,
+            phase_index: 0.0,
+        }
+    }
+    /// Set the frequency of the LFO in Hz.
+    /// Example usage
+    /// ```
+    /// let mut lfo = LFO::new(44100.0, 1024);
+    /// lfo.set_frequency(1.0);
+    /// assert_eq!(lfo.freq_hz, 1.0);
+    /// ```
+    pub fn set_frequency(&mut self, freq_hz: f32) {
+        self.freq_hz = freq_hz;
+    }
+
+    /// Set the Amplitude of LFO wavetable.
+    /// Example usage
+    /// ```
+    /// let mut lfo = LFO::new(44100.0, 1024);
+    /// lfo.set_amplitude(2.0);
+    /// assert_eq!(lfo.amplitude, 2.0);
+    /// ```
+    pub fn set_amplitude(&mut self, amplitude: f32) {
+        self.amplitude = amplitude;
+    }
+
+    pub fn set_phase(&mut self, phase: f32) {
+        self.phase_index = phase;
+    }
+
+    pub fn get_params(&self) -> (f32, f32, f32) {
+        (self.freq_hz, self.amplitude, self.phase_index)
+    }
+    /// Reset the phase index of the LFO.
+    /// Example usage
+    /// ```
+    /// let mut lfo = LFO::new(44100.0, 1024);
+    /// lfo.reset(2048);
+    /// assert_eq!(lfo.phase_index, 0.0);
+    pub fn reset(&mut self, size: usize) {
+        self.phase_index = 0.0;
+        self.wave_table = RingBuffer::<f32>::new(size);
+        // very inefficient, but cannot come up with a better way
+        for i in 0..size {
+            let phase = (i as f32 / size as f32) * 2.0 * PI;
+            self.wave_table.push(phase.sin());
+        }
+    }
+
+    pub fn next_mod(&mut self) -> f32 {
```

Comment:
Would be helpful to add a comment or two about next_mod, since this is also called in vibrato.rs